### DISCUSSION TITLE
POST Api is cancelling, it is intercepted and it got cancelled

### DISCUSSION THREAD

**Discussion by AkhilPatell** [cypress-io/cypress#28503] [Questions and Help]
Why POST Api is cancelling, it is intercepted but it got cancelled.

cy.fixture("temp_opportunity").then((team_competition) => {
        cy.reload()
          cy.wait(1000);
          // Change Leader
            //Intercept the API call
            cy.intercept('POST', `/api/v1/competition/${team_competition.teamCode}/teamMembersDetails`).as('apiCall2');
          cy.get('[class="ki ki-bold-more-hor"]').should('exist')
              .click({ force: true });
          cy.get('[class="navi-text"]').should('exist')
              .and('contain', "Change Leader")
              .eq(1).click({ force: true }); 
          cy.get('[class="m-2 text-gray-400 fw-bold fs-5"]').should('exist')
              .and('contain', "I confirm")
          cy.get('[class="mat-button-wrapper"]').should('exist')
              .and('contain', "Submit")
              .click({ multiple: true, force: true });
            cy.wait(1000);
          cy.wait("@apiCall2").then((interception) => {
              expect(interception.response.statusCode).to.be.oneOf[200,301,304];
          });
          
          having two api with same headers
          but GET, and POST methods respectively
          but POST is cancelling
          
<img width="1470" alt="Screenshot 2023-12-12 at 2 55 18 PM" src="https://github.com/cypress-io/cypress/assets/116702495/de50a9e2-0084-46fb-95fd-60f04f436a93">
          
<img width="1470" alt="Screenshot 2023-12-12 at 2 55 02 PM" src="https://github.com/cypress-io/cypress/assets/116702495/411706b0-e6ba-4d96-81f4-845aa693c829">

<img width="1470" alt="Screenshot 2023-12-12 at 2 55 10 PM" src="https://github.com/cypress-io/cypress/assets/116702495/9888cb77-5e14-45e2-bcd9-6b744a94b0a0">

PS: please help me out in this situation
as backend code is correct.

**Comment by portyu9** [comment]
Hi @AkhilPatell 

This intercept by itself does **not** cancel the request:

```js
cy.intercept(
  'POST',
  `/api/v1/competition/${team_competition.teamCode}/teamMembersDetails`
).as('apiCall2')
```

With no route handler or static response, `cy.intercept()` is only observing the request.

The first thing I would change is this:

```js
cy.get('[class="mat-button-wrapper"]')
  .should('exist')
  .and('contain', 'Submit')
  .click({ multiple: true, force: true })
```

`multiple: true` tells Cypress to click **every matching element**. Combined with a broad class selector and `force: true`, that can produce multiple submissions/navigation events and can absolutely complicate an in-flight request.

Target one actual Submit button and let Cypress perform normal actionability checks:

```js
cy.intercept(
  'POST',
  `/api/v1/competition/${team_competition.teamCode}/teamMembersDetails`
).as('apiCall2')

cy.contains('button', 'Submit')
  .should('be.enabled')
  .click()

cy.wait('@apiCall2').then(({ response, error }) => {
  expect(error).to.be.undefined
  expect(response, 'response').to.exist

  expect(response.statusCode).to.be.oneOf([200, 301, 304])
})
```

Also note that the original assertion syntax is incorrect:

```js
expect(interception.response.statusCode).to.be.oneOf[200,301,304]
```

`oneOf` is a function:

```js
expect(interception.response.statusCode).to.be.oneOf([200, 301, 304])
```

I would remove the fixed `cy.wait(1000)` calls as well. The aliased network request is the deterministic thing to wait for.

If the request is still shown as `(canceled)` after making the Submit action singular, inspect whether:

- the Submit action causes immediate page navigation;
- application code calls `AbortController.abort()`;
- two identical POSTs are being created;
- the component/page is being destroyed while the request is active.

There is nothing in the shown `cy.intercept()` that would itself cancel the request.

Please mark this answer as accepted if it helped, thank you.
